### DISCUSSION TITLE
Event-Driven NETCONF Subscription Support in scrapligo

### DISCUSSION THREAD

**Discussion by bruckynetworkstest** [scrapli/scrapligo#161] [Q&A] [answered]
Hi, it's been a while, I've been eager beavering away with the library.

In ncclient, there's a create_subscription method that allows for establishing an event-driven NETCONF subscription to receive notifications upon changes. 

Is there a way to establish an event-driven NETCONF subscription in scrapligo similar to what create_subscription provides in ncclient?

From what I can make out (which is certainly not definitive) it seems the library only supports periodic subscriptions via the EstablishPeriodicSubscription method?

**Comment by carlmontanari** [comment, marked as answer]
quickly looking it looks like the only difference is not having `period` in the payload? if thats true then probably would be easy to tweak ether adding a new driver method or making the current method more generic, but I think the answer in short is "no".

**Reply by bruckynetworkstest** [reply]
Thanks as always for quick and precise response. 

All the best ...
